join-statements/joinstatements.md

```diff
@@ -0,0 +1,423 @@
+FROM students
+FULL OUTER JOIN societies ON students.id = societies.student_id;
+```
+
+3) **Primary key** would be a key that is a unique identifier in a table, and is only found once within the entire table. This would allow you to identify a single row of data just by it's **primary key**.
+
+A **primary key** real world example would be in a college, your student ID #. This would be your primary key in the College's `students` database table.
+
+A **foreign key** would be a key in a table that links entries from this table to another table via the other table's *primary key*. Typically used in many to one relationships where we could link multiple of the same *primary keys* in a table by assigning it to a *foreign key* value.
+
+A **foreign key** real world example would be your major's program ID found within your Colleges *students* table. Alternatively, the foreign key could also be a society that the student belongs to. The major ID or society ID would be the *primary key* in the College's `program major` table or the `societies` table respectively. The major ID or society ID could both be **foreign keys** in the `students` table.
+
+4) Aliasing is utilizing a different variable name to represent a table's name or a column's name. Typically done by using the statement `AS` when declaring a `FROM` or `JOIN` statement for  representing a table name, such as:
+
+> SELECT t.column1 
+> FROM table_name AS t
+> WHERE t.id = 20
+> INNER JOIN table2_name;
+
+or to represent an aliased column:
+
+> SELECT AVG(column1) AS average_score, name
+> FROM table_name
+> WHERE alias1 = 20
+> GROUP BY name
+> ORDER BY average_score DESC;
+
+This would allow for more readable queries, and perhaps economize keystrokes for typing if the query is particularly long and the aliased column or table is used multiple times in the query.
+
+5) 
+SELECT p.name, c.salary, c.vacation_days 
+FROM professor AS p JOIN
+compensation AS c ON p.id = c.professor_id;
+
+6)
+`NATURAL JOIN` would be ideal to use in cases where column names have the exact same name in both tables. For example, if we have a `fantasy_gamer` table with the the columns `real_name`, `email_address`, and another table called `charachters` with the columns `charachter_name` and `email_address`, we would be able to create the following query:
+
+```
+SELECT fantasy_gamer.real_name
+FROM fantasy_gamer
+NATURAL JOIN charachters;
+
+```
+
+this query would yield the real names of users with emails that match in both `fantasy_gamer` table and the `charachters` table.
+
+This is because there exists an `email_address` column in both the `fantasy_gamer` table, as well as the `charachters` table.
+
+7) 
+
+I created these tables:
+```
+contract_business=# SELECT *
+contract_business-# FROM shifts
+contract_business-# ORDER BY shift_id ;
+
+ shift_id | day_of_week | shift | employee_id 
+----------+-------------+-------+-------------
+        0 | Sunday      |     0 |           3
```

Comment:
the intent was for these to be unassigned

Reply:
done